packages/mask/src/plugins/Savings/SNSAdaptor/SavingsDialog.tsx

```diff
@@ -0,0 +1,113 @@
+import { useState } from 'react'
+import { useAsync } from 'react-use'
+import { DialogContent } from '@mui/material'
+import { ChainId, getChainIdFromNetworkType, useChainId } from '@masknet/web3-shared-evm'
+import { isDashboardPage } from '@masknet/shared-base'
+import { useI18N } from '../../../utils'
+import { InjectedDialog } from '../../../components/shared/InjectedDialog'
+import { WalletStatusBox } from '../../../components/shared/WalletStatusBox'
+import { FolderTabPanel, FolderTabs } from '@masknet/theme'
+import { NetworkTab } from '../../../components/shared/NetworkTab'
+import { WalletRPC } from '../../Wallet/messages'
+import { ProtocolCategory, ProtocolType, TabType } from '../types'
+import { SavingsProtocols } from '../protocols'
+import { useStyles } from './SavingsDialogStyles'
+import { SavingsTable } from './SavingsTable'
+import { SavingsForm } from './SavingsForm'
+
+interface SavingsDialogProps {
+    open?: boolean
+    onClose?: () => void
+    onSwapDialogOpen?: () => void
+}
+
+export function SavingsDialog({ open, onClose, onSwapDialogOpen }: SavingsDialogProps) {
+    const { t } = useI18N()
+    const isDashboard = isDashboardPage()
+    const { classes } = useStyles({ isDashboard })
+
+    const currentChainId = useChainId()
+    const [chainId, setChainId] = useState<ChainId>(currentChainId)
+    const [tab, setTab] = useState<TabType>(TabType.Deposit)
+    const [selectedProtocol, setSelectedProtocol] = useState<ProtocolType | null>(null)
+
+    const { value: chains } = useAsync(async () => {
```

Comment:
```suggestion
    const { value: chains = EMPTY_LIST } = useAsync(async () => {
```

`import { EMPTY_LIST } from '@masknet/web3-shared-evm`


Use a stable array to reduce re-rendering.